content/dev-tools/rpc-tool.page.tsx

```diff
@@ -0,0 +1,552 @@
+
+  const FULL_HISTORY_SERVER = "wss://s2.ripple.com";
+  const reTxId = /^[0-9A-Fa-f]{64}$/;
+  const reLedgerSeq = /^[0-9]+$/;
+
+  /*
+    XRPL requests
+    * account_info
+    * account_lines
+    * account_tx
+    * account_objects
+    * tx
+    * ledger
+  */
+
+  const accountInfo = async (
+    // @ts-expect-error - xrpl is added via a script tag
+    client: xrpl.Client,
+    address: string
+  ): Promise<void> => {
+    const response = await client.request({
+      command: "account_info",
+      account: address,
+    });
+    setProgressBarWidth("20%");
+    setAccountInfoResponse(response);
+  };
+
+  const accountLines = async (
+    // @ts-expect-error - xrpl is added via a script tag
+    client: xrpl.Client,
+    address: string
+  ): Promise<void> => {
+    const response = await client.request({
+      command: "account_lines",
+      account: address,
+    });
+    setProgressBarWidth("40%");
+    setAccountLinesResponse(response);
+  };
+
+  const accountTx = async (
+    // @ts-expect-error - xrpl is added via a script tag
+    client: xrpl.Client,
+    address: string
+  ): Promise<void> => {
+    const response = await client.request({
+      command: "account_tx",
+      account: address,
+      ledger_index_min: -1,
+      ledger_index_max: -1,
+      binary: false,
+      limit: 20,
+      forward: false,
+    });
+    setProgressBarWidth("60%");
+    setAccountTxResponse(response);
+  };
+
+  const accountObjects = async (
+    // @ts-expect-error - xrpl is added via a script tag
+    client: xrpl.Client,
+    address: string
+  ): Promise<void> => {
+    const response = await client.request({
+      command: "account_objects",
+      account: address,
+    });
+    setProgressBarWidth("80%");
+    setAccountObjectsResponse(response);
+  };
+
+  const tx = async (
+    // @ts-expect-error - xrpl is added via a script tag
+    client: xrpl.Client,
+    transactionId: string
+  ): Promise<void> => {
+    const response = await client.request({
+      command: "tx",
+      transaction: transactionId,
+      binary: false,
+    });
+    setProgressBarWidth("100%");
+    setTxResponse(response);
+  };
+
+  const ledger = async (
+    // @ts-expect-error - xrpl is added via a script tag
+    client: xrpl.Client,
+    ledgerSequence: string
+  ): Promise<void> => {
+    const response = await client.request({
+      command: "ledger",
+      ledger_index: ledgerSequence,
+      transactions: true,
+      expand: true,
+    });
+    setProgressBarWidth("100%");
+    setLedgerResponse(response);
+  };
+
+  useEffect(() => {
+    if (slug && slug !== "") {
+      getInfo();
+    }
+  }, []);
+
+  const getInfo = async (): Promise<void> => {
+    setAccountInfoResponse(null);
+    setAccountLinesResponse(null);
+    setAccountTxResponse(null);
+    setAccountObjectsResponse(null);
+    setTxResponse(null);
+    setLedgerResponse(null);
+    setErrorText(null);
+    setShowResult(true);
+    setProgressBarWidth("0%");
```

Comment:
Can you group these variables with comments explaining the groupings? (Some of these seem like they should be broken out into sub-components, and some seem like separate parts of the page)

Reply:
If you make sub-components, please add them to a sub-folder :)

Reply:
I think this has been done with my refactor.